src/App.jsx

```diff
@@ -14,33 +16,90 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_BY_ALPHABET = 'Sort alphabetically';
+const SORT_BY_LENGTH = 'Sort by length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+export const App = () => {
+  const [visibleGoods, setVisibleGoods] = useState(goodsFromServer);
+  const [sort, setSort] = useState({ type: '', reversed: false });
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  // function for sort by length and alphabet
+  const handleSort = type => {
+    const sortedGoods = [...goodsFromServer];
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    if (type === SORT_BY_ALPHABET) {
+      sortedGoods.sort((a, b) => a.localeCompare(b));
+    } else if (type === SORT_BY_LENGTH) {
+      sortedGoods.sort((a, b) => a.length - b.length);
+    }
+
+    if (sort.reversed) {
+      sortedGoods.reverse();
+    }
+
+    setSort({ type, reversed: sort.reversed });
+    setVisibleGoods(sortedGoods);
+  };
+
+  // function for reverse
+  const handleReverse = () => {
+    const reversedGoods = [...visibleGoods].reverse();
+
+    setSort(prev => ({ ...prev, reversed: !prev.reversed }));
+    setVisibleGoods(reversedGoods);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  // function for reset
+  const handleReset = () => {
+    setSort({ type: '', reversed: false });
+    setVisibleGoods(goodsFromServer);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button ${sort.type === SORT_BY_ALPHABET ? '' : 'is-light'}`}
+          onClick={() => handleSort(SORT_BY_ALPHABET)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={`button is-success ${sort.type === SORT_BY_LENGTH ? '' : 'is-light'}`}
+          onClick={() => handleSort(SORT_BY_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={`button is-info ${sort.reversed ? '' : 'is-light'}`}
+          onClick={handleReverse}
+        >
+          Reverse
+        </button>
+
+        {sort.type || sort.reversed ? (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={handleReset}
+          >
+            Reset
+          </button>
+        ) : null}
```

Comment:
You don't need the ternary operator if you return nothing in the second case, use `&&` instead